pkg/server/redirect.go

```diff
@@ -0,0 +1,61 @@
+package server
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/drone/drone/Godeps/_workspace/src/github.com/gin-gonic/gin"
+)
+
+// RedirectSha accepts a request to retvie a redirect
+// to job from the datastore for the given repository
+// and commit sha
+//
+//  GET /redirect/:owner/:name/commit/:sha
+//
+func RedirectSha(c *gin.Context) {
+	var branch string
+
+	store := ToDatastore(c)
+	repo := ToRepo(c)
+	sha := c.Params.ByName("sha")
+
+	branch = c.Request.FormValue("branch")
+	if branch == "" {
+		branch = repo.Branch
+	}
+
+	build, err := store.BuildSha(repo, sha, branch)
+	if err != nil {
+		c.Redirect(301, "/")
+		return
+	}
+
+	c.Redirect(301, fmt.Sprintf("/%s/%s/%d", repo.Owner, repo.Name, build.ID))
```

Comment:
I added some notes about slightly altering the route. Since we have the build, let's just to `c.Json` instead of `c.Redirect` since I think this may be a userful endpoint for others as well